x-pack/legacy/plugins/siem/public/lib/compose/kibana_compose.tsx

```diff
@@ -4,28 +4,19 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { InMemoryCache, IntrospectionFragmentMatcher } from 'apollo-cache-inmemory';
-import ApolloClient from 'apollo-client';
-import { ApolloLink } from 'apollo-link';
+import { ApolloClient, ApolloLink, InMemoryCache } from '@apollo/client';
 
-import introspectionQueryResultData from '../../graphql/introspection.json';
 import { CoreStart } from '../../plugin';
 import { AppFrontendLibs } from '../lib';
 import { getLinks } from './helpers';
 
 export function compose(core: CoreStart): AppFrontendLibs {
-  const cache = new InMemoryCache({
-    dataIdFromObject: () => null,
-    fragmentMatcher: new IntrospectionFragmentMatcher({
-      introspectionQueryResultData,
-    }),
-  });
   const basePath = core.http.basePath.get();
 
   const apolloClient = new ApolloClient({
     connectToDevTools: process.env.NODE_ENV !== 'production',
-    cache,
-    link: ApolloLink.from(getLinks(cache, basePath)),
+    cache: new InMemoryCache(),
+    link: ApolloLink.from(getLinks(basePath)),
```

Comment:
You might be ok here, but there are some steps from this PR and the reasoning behind why we had the:
```ts
dataIdFromObject: () => null,
fragmentMatcher: new IntrospectionFragmentMatcher({	
      introspectionQueryResultData,	
}),
```

Very long explanation and test steps here:
https://github.com/elastic/kibana/pull/30180

We might be ok with things the way they are, but if you see anything such as "disappearing data" on drag and drops or other things going on that would be the why. I would still follow the steps from that PR and look at the Apollo cache for testing here to ensure the in-memory cache operates with keys and values the way you expect it to.